ledfx/events.py

```
@@ -34,6 +36,38 @@ def to_dict(self):
+    """Event emitted with arbitrary test diagnostics"""
+
+    def __init__(self, debug: str, scroll: bool = False):
+        super().__init__(Event.GENERAL_DIAG)
+        self.debug = debug
+        self.scroll = scroll
+
+
+class VirtualDiagEvent(Event):
+    """Event emitted when a virtual's diagnostics are updated"""
+
+    def __init__(
```

Pylint (3.3.7)

[refactor] Too many arguments (8/5)

(R0913)